examples/pytorch/schnet/Alchemy_dataset.py

```diff
@@ -0,0 +1,224 @@
+                          Chem.rdchem.BondType.AROMATIC, "NoBond")
+            ]
+            e[(n1, n2)] = e_t
+        nx.set_edge_attributes(g, e, "e_feat")
+
+    # sdf file reader for Alchemy dataset
+    def sdf_graph_reader(self, sdf_file):
+
+        with open(str(sdf_file), 'r') as f:
+            sdf_string = f.read()
+        mol = Chem.MolFromMolBlock(sdf_string, removeHs=False)
+        if mol is None:
+            print("rdkit can not parsing", sdf_file)
+            return None
+        feats = self.chem_feature_factory.GetFeaturesForMol(mol)
+
+        g = nx.DiGraph()
+        l = torch.FloatTensor(self.target.loc[int(sdf_file.stem)].tolist()) \
+            if self.mode == 'dev' else torch.LongTensor([int(sdf_file.stem)])
+
+        # Create nodes
+        assert len(mol.GetConformers()) == 1
+        geom = mol.GetConformers()[0].GetPositions()
+        for i in range(mol.GetNumAtoms()):
+            atom_i = mol.GetAtomWithIdx(i)
+            g.add_node(i,
+                       node_str=atom_i.GetSymbol(),
+                       node_type=atom_i.GetAtomicNum(),
+                       acceptor=0,
+                       donor=0,
+                       aromatic=atom_i.GetIsAromatic(),
+                       hybridization=atom_i.GetHybridization(),
+                       num_h=atom_i.GetTotalNumHs(),
+                       pos=torch.FloatTensor(geom[i]))
+
+        for i in range(len(feats)):
+            if feats[i].GetFamily() == 'Donor':
+                node_list = feats[i].GetAtomIds()
+                for i in node_list:
+                    g.node[i]['donor'] = 1
+            elif feats[i].GetFamily() == 'Acceptor':
+                node_list = feats[i].GetAtomIds()
+                for i in node_list:
+                    g.node[i]['acceptor'] = 1
+        # Read Edges
+        for i in range(mol.GetNumAtoms()):
+            for j in range(mol.GetNumAtoms()):
+                e_ij = mol.GetBondBetweenAtoms(i, j)
+                # remove self-loop
+                if i == j:
+                    continue
+                if e_ij is not None:
+                    g.add_edge(i, j, b_type=e_ij.GetBondType())
+                else:
+                    g.add_edge(i, j, b_type="NoBond")
+                g.add_edge(i, j, distance=np.linalg.norm(geom[i] - geom[j]))
+
+        self.alchemy_nodes(g)
+        self.alchemy_edges(g)
+        ret = dgl.DGLGraph()
+        ret.from_networkx(g,
+                          node_attrs=['n_feat', 'pos', 'node_type'],
+                          edge_attrs=['e_feat', 'distance'])
+        ret.edata["distance"] = ret.edata["distance"].reshape(-1, 1)
+
+        return ret, l
+
+    def __init__(self, mode='dev', transform=None):
+        assert mode in ['dev', 'valid',
+                        'test'], "mode should be dev/valid/test"
+        self.mode = mode
+        self.transform = transform
+        self.file_dir = pathlib.Path(get_download_dir(), mode)
```

Comment:
Note that you do not need to append the directory with mode. With `pathlib.Path(get_download_dir(), mode)`, we will get an error `File b'/.dgl/dev/train.csv'` does not exist as the path for the file in fact becomes `File b'/.dgl/dev/dev/train.csv'`.